src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023holostiminterface.py

```diff
@@ -0,0 +1,242 @@
+    ) -> None:
+        metadata_copy = deepcopy(metadata)
+
+        device_name = metadata_copy["Ophys"]["Device"][0]["name"]
+        device = nwbfile.devices[device_name]
+
+        # Add stimulus pattern to NWBFile
+        temporal_focusing_metadata = metadata_copy["LabMetaData"]["TemporalFocusing"]
+        temporal_focusing = TemporalFocusing(**temporal_focusing_metadata)
+        nwbfile.add_lab_meta_data(temporal_focusing)
+
+        # Add SLM device to NWBFile
+        spatial_light_modulator_metadata = metadata_copy["Ophys"]["OptogeneticDevice"]["SpatialLightModulator2D"]
+        spatial_light_modulator_metadata.update(spatial_resolution=[1024, 1024])
+        spatial_light_modulator_name = spatial_light_modulator_metadata["name"]
+        if spatial_light_modulator_name not in nwbfile.devices:
+            nwbfile.add_device(SpatialLightModulator2D(**spatial_light_modulator_metadata))
+        spatial_light_modulator = nwbfile.devices[spatial_light_modulator_name]
+
+        # Add light source device to NWBFile
+        light_source_metadata = metadata_copy["Ophys"]["OptogeneticDevice"]["LightSource"]
+        light_source_name = light_source_metadata["name"]
+        if light_source_name not in nwbfile.devices:
+            nwbfile.add_device(LightSource(**light_source_metadata))
+        light_source = nwbfile.devices[light_source_name]
+
+        # Add optogenetic stimulus site to NWBFile
+        stimulus_site_metadata = metadata_copy["Ophys"]["OptogeneticStimulusSite"][0]
+        stimulus_site_metadata.update(
+            device=device, spatial_light_modulator=spatial_light_modulator, light_source=light_source
+        )
+        stim_site = PatternedOptogeneticStimulusSite(**stimulus_site_metadata)
+        nwbfile.add_ogen_site(stim_site)
+
+        # Create a plane segmentation object for targeted ROIs
+        optical_channel = OpticalChannel(
+            name="Chan1",
+            description="Green channel of the microscope.",
+            emission_lambda=500.0,
+        )
+
+        fov_size_in_um = np.array(self.rois_metadata["imagingRoiGroup"]["rois"]["scanfields"]["sizeXY"])
+        frame_dimesion = np.array(self.rois_metadata["imagingRoiGroup"]["rois"]["scanfields"]["pixelResolutionXY"])
+
+        imaging_plane_name = "ImagingPlaneHolographicStimulation"
+        imaging_plane = nwbfile.create_imaging_plane(
+            name=imaging_plane_name,
+            optical_channel=optical_channel,
+            imaging_rate=float(
+                self.image_metadata["SI.hRoiManager.scanFrameRate"]
+            ),  # Frame rate on the entire volumetric stack
+            description="Imaging plane for the holographic stimulation",
+            device=device,
+            excitation_lambda=600.0,  # TODO update
+            indicator="GFP",  # TODO update
+            location="V1",  # TODO update
+            grid_spacing=fov_size_in_um / frame_dimesion,
+            grid_spacing_unit="micrometers",
```

Comment:
Is there a reason why not use the utility method `add_imaging_plane() `from neuroconv for this?

Reply:
I got this error:
`The `._to_dict()` method for pickling has not been defined for this DataChunkIterator!`

Reply:
yeah just make sure you're not passing objects to metadata `dict`, e.g. for optical channel if this is a metadata update then it should be the name of the OpticalChannel, same for device which should be the name of the Device.